earth2studio/data/goes.py

```diff
@@ -0,0 +1,659 @@
+            except IndexError:
+                return False
+
+        except Exception:
+            return False
+
+    @classmethod
+    def grid(
+        cls, satellite: str = "goes16", scan_mode: str = "F"
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """
+        Return (lat, lon) in degrees for the native GOES grid using pyproj.
+
+        Parameters
+        ----------
+        satellite : str, optional
+            Which GOES satellite to use, by default "goes16"
+        scan_mode : str, optional
+            Scan mode ('F' for Full Disk, 'C' for Continental US), by default "F"
+
+        Notes
+        -----
+        This function is based on the GOES ABI fixed grid projection variables and constants.
+        The projection comes from the recommended NOAA documentation:
+        https://www.star.nesdis.noaa.gov/atmospheric-composition-training/python_abi_lat_lon.php
+
+        Returns
+        -------
+        tuple[np.ndarray, np.ndarray]
+            Tuple of (lat, lon) in degrees
+        """
+
+        # Read in GOES ABI fixed grid projection variables and constants
+        if scan_mode == "F":
+            x_coordinate_1d = GOES.FULL_DISK_YX[1]  # E/W scanning angle in radians
+            y_coordinate_1d = GOES.FULL_DISK_YX[0]  # N/S elevation angle in radians
+        else:
+            x_coordinate_1d = GOES.CONTINENTAL_US_YX[satellite][
+                1
+            ]  # E/W scanning angle in radians
+            y_coordinate_1d = GOES.CONTINENTAL_US_YX[satellite][
+                0
+            ]  # N/S elevation angle in radians
+        lon_origin = GOES.LONGITUDE_OF_PROJECTION_ORIGIN[satellite]
+        H = GOES.PERSPECTIVE_POINT_HEIGHT + GOES.SEMI_MAJOR_AXIS
+        r_eq = GOES.SEMI_MAJOR_AXIS
+        r_pol = GOES.SEMI_MINOR_AXIS
+
+        # Create 2D coordinate matrices from 1D coordinate vectors
+        x_coordinate_2d, y_coordinate_2d = np.meshgrid(x_coordinate_1d, y_coordinate_1d)
+
+        # Equations to calculate latitude and longitude
+        lambda_0 = (lon_origin * np.pi) / 180.0
+        a_var = np.power(np.sin(x_coordinate_2d), 2.0) + (
+            np.power(np.cos(x_coordinate_2d), 2.0)
+            * (
+                np.power(np.cos(y_coordinate_2d), 2.0)
+                + (
+                    ((r_eq * r_eq) / (r_pol * r_pol))
+                    * np.power(np.sin(y_coordinate_2d), 2.0)
+                )
+            )
+        )
+        b_var = -2.0 * H * np.cos(x_coordinate_2d) * np.cos(y_coordinate_2d)
+        c_var = (H**2.0) - (r_eq**2.0)
+        r_s = (-1.0 * b_var - np.sqrt((b_var**2) - (4.0 * a_var * c_var))) / (
+            2.0 * a_var
+        )
+        s_x = r_s * np.cos(x_coordinate_2d) * np.cos(y_coordinate_2d)
+        s_y = -r_s * np.sin(x_coordinate_2d)
+        s_z = r_s * np.cos(x_coordinate_2d) * np.sin(y_coordinate_2d)
+
+        # Ignore numpy errors for sqrt of negative number; occurs for GOES-16 ABI CONUS sector data
+        np.seterr(all="ignore")
```

Comment:
Does this set this package wide?
Is there a way to unset this at the end of the function?